### DISCUSSION TITLE
AutoRelay?

### DISCUSSION THREAD

**Discussion by Winterhuman** [alecbcs/hyprspace#39] [Q&A] [answered]
Is hyprspace's node configured with AutoRelay and/or RelayHop enabled? I looked through the code and couldn't find any mention of the settings.

**Comment by alecbcs** [comment, marked as answer]
@LynHyper at the moment we don't use libp2p's `AutoRelay` or `RelayHop` feature. Not to say that we couldn't add this in the future but, as far as I understand their current implementation within libp2p, libp2p only makes use of a set of hardcoded known relays by default. My worry in implementing this is then having the libp2p relays change outside of hyprspace and some users dependent on this feature losing their connections until we update libp2p and publish a new version of hyprspace.

I'll admit this is a bit hypocritical since we rely on the libp2p discovery nodes at the moment. However, I would like to change this soon so that power users can define one of their own hyprspace instances with a public IP to be network's discovery node.

**Comment by Winterhuman** [comment]
AutoRelay can now auto-discover CircuitV2 relays as of go-libp2p v0.19 (https://github.com/libp2p/go-libp2p/releases/tag/v0.19.0), this means relays are no longer hardcoded.
